### PR TITLE
Fix "Click interactions require two clicks #553"

### PR DESCRIPTION
**Pull request by timothyfroehlich**
I'm not confident that this is the best fix, but it does seem to work. By making the coroutine start undispatched it runs immediately and handles the click properly again. 

I've added a unit test to exercise some of the interaction tests. More can be added later.

### CODE REVIEW

**Comment by rylin8** [comment]
Added @iamralpht since he wrote this and knows it better.

File: designcompose/src/main/java/com/android/designcompose/DesignView.kt [hunk 1]
```diff
@@ -502,7 +503,7 @@ internal fun DesignView(
                     // event so that the interaction state can revert the change).
                     //
                     // The interaction test document covers all of these cases.
-                    interactionScope.launch {
+                    interactionScope.launch(start = CoroutineStart.UNDISPATCHED) {
```
Comment (iamralpht) [inline designcompose/src/main/java/com/android/designcompose/DesignView.kt:506]: I'm not sure why we need `interactionScope` to start with. I did write it, but I don't remember why. Reading the documentation on `CoroutineStart` it looks like the default behavior is sometimes lazy, which I guess explains why the coroutine wasn't getting started immediately... but I still don't know what the point of having an explicit coroutine scope is.

Comment (timothyfroehlich) [reply designcompose/src/main/java/com/android/designcompose/DesignView.kt:506]: My best guess is that this was trying to get the interaction to survive the destruction of the Composable and replacement with one of its variants. Perhaps we should get on GVC to explore it more?

Comment (iamralpht) [reply designcompose/src/main/java/com/android/designcompose/DesignView.kt:506]: Ah, yes, that does make sense; the drag and press-and-hold interactions need to be kept alive after they've started, even though the first part of the interaction might remove the Composable that's holding the interaction itself.

**Review by iamralpht — APPROVED** [review]

**Pull request merged**